lib/fog/compute/ecloud/models/ssh_key.rb

```diff
@@ -9,6 +9,30 @@ class SshKey < Fog::Ecloud::Model
         attribute :other_links, :aliases => :Links
         attribute :default, :aliases => :Default, :type => :boolean
         attribute :finger_print, :aliases => :FingerPrint
+        attribute :private_key, :aliases => :PrivateKey
+
+        def delete
+          service.ssh_key_delete(href).body
+        end
+        alias_method :destroy, :delete
+
+        def edit(options = {})
+          # Make sure we only pass what we should
+          new_options = {}
+          new_options[:uri] = href
+          if options[:name].nil?
+            new_options[:name] = name
+          else
+            new_options[:name] = options[:name]
+          end
+          if options[:default].nil?
+            new_options[:default] = default
+          else
+            new_options[:default] = options[:default]
+          end
```

Comment:
Why are you checking for nil values here?

If all this method accepts are `uri`, `name` and `default`, why not have a different signature on the request?

```service.ssh_key_edit(uri, name, default).body```